azure-shared-resource.go

```diff
@@ -1,5 +1,7 @@
 package batcher
```

Comment:
I'm having trouble understanding how the `count < target` condition within the Start() operation will ever be true because the sample code which demonstrates how the rate limiter and batcher work together has an azureSharedResource instance that does not initialize the rate limiter's `target` field with a value  nor the rate limiter's `partitions` field with values. 

Sample code demonstrates the batcher initializing the fields sometime after the azureSharedResource start() function has already been called:
```
if err := azresource.Provision(ctx); err != nil {
		panic(err)
	}
	if err := azresource.Start(ctx); err != nil {
		panic(err)
	}

	// configure the batcher
	batcher := gobatcher.NewBatcher().
		WithRateLimiter(azresource)
```

Reply:
I think you are missing a couple of key things:

- Calling the Start() func begins the rate limiter's processing loop. It is generally true when this is first run that the count and target are 0, so nothing happens.

- However, as soon as the Batcher has some work to do, it calls GiveMe() which increases the target. This is done at the CapacityInterval.

- Generally, the rate limiter's processing loop runs several times a second and it will now see that count is less than target and will attempt to allocate a partition.

Reply:
Should've hit the ingest endpoint sooner..

![image](https://user-images.githubusercontent.com/10041279/103059233-24babd80-456a-11eb-911f-ae87e50ce0cc.png)

Reply:
I am not sure what this screenshot is supposed to be for. Are you suggesting that it is working correctly or not working correctly?

Reply:
Working correctly. Just sharing that GiveMe() increased the target for me. In this case, by 20.